src/components/item/publish/useItemLicense.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState } from 'react';
+
+import { SxProps } from '@mui/material';
+
+import { DiscriminatedItem } from '@graasp/sdk';
+import { CCSharingVariant, CreativeCommons } from '@graasp/ui';
+
+import { mutations } from '@/config/queryClient';
+import { convertLicense, convertSelectionToLicense } from '@/utils/itemLicense';
+
+import LicenseForm from './LicenseForm';
+import { CCLicenseChoice, CCSharingLicenseChoice } from './type';
+
+const licensePreviewStyle = {
+  border: '1px solid #eee',
+  borderRadius: 2,
+  minWidth: 300,
+};
+
+const useItemLicense = ({
+  item,
+  disabled,
+  commonsSx,
+}: {
+  item: DiscriminatedItem;
+  disabled?: boolean;
+  commonsSx?: SxProps;
+}): {
+  handleSubmit: () => void;
+  licenseForm: JSX.Element;
+  creativeCommons: JSX.Element;
+  requireAttributionValue: CCLicenseChoice;
+} => {
+  const [requireAttributionValue, setRequireAttributionValue] =
+    useState<CCLicenseChoice>('');
+  const [allowCommercialValue, setAllowCommercialValue] =
+    useState<CCLicenseChoice>('');
+  const [allowSharingValue, setAllowSharingValue] =
+    useState<CCSharingLicenseChoice>('');
+  // itemId
+  const itemId = item?.id;
+
+  const settings = item?.settings;
+  const itemName = item?.name;
+
+  const { mutate: updateCCLicense } = mutations.useEditItem();
+
+  useEffect(() => {
+    if (settings?.ccLicenseAdaption) {
+      const { allowCommercialUse, allowSharing, requireAccreditation } =
+        convertLicense(settings.ccLicenseAdaption);
+      setAllowSharingValue(allowSharing);
+      setAllowCommercialValue(allowCommercialUse ? 'yes' : 'no');
+      setRequireAttributionValue(requireAccreditation ? 'yes' : 'no');
+    }
+  }, [settings]);
+
+  const handleSubmit = () => {
+    if (requireAttributionValue) {
+      updateCCLicense({
+        id: itemId,
+        name: itemName,
```

Comment:
Do you have to specify the name when you update the item settings ??

Reply:
No need at all